Sync Properties Update Revision Block and Create PDF/Modules/Sample.SyncFilePropsAndUpdRevBlockModule.psm1

```diff
@@ -0,0 +1,846 @@
+			}
+		}
+	}
+'@
+#endregion
+
+function SyncProperties
+{
+    param($file)
+
+	Write-Host "Start 'Sync Properties' of file '$($file.Name)' ..."
+	$s = "/AutodeskDM/Services/v"
+	$start = $vault.AdminService.Url.IndexOf($s) + $s.Length
+	$version = [int]$vault.AdminService.Url.Substring($start, 2)
+
+	# 27 = 2022
+	# 25 = 2020
+	# 24 = 2019
+	# 23 = 2018
+	# 22 = 2017
+	# 21 = 2016
+	
+	if ($version -ge 27) {
+		$cs = $2022AndNewerCs
+	} elseif ($version -ge 25) {
+		$cs = $2020AndNewerCs
+	} elseif ($version -ge 22) {
+		$cs = $2017AndNewerCs
+	} else {
+		$cs = $OlderThan2017Cs
+	}
+
+	$null = [System.Reflection.Assembly]::LoadWithPartialName("Autodesk.Connectivity.WebServices")
+	Add-Type -TypeDefinition $cs -ReferencedAssemblies @("Autodesk.Connectivity.WebServices", "System.Core")
+	
+	$propertySync = New-Object Autodesk.Vault.SDKTools.PropertySync -ArgumentList @($vault)
+	$writeResults = New-Object Autodesk.Connectivity.WebServices.PropWriteResults
+	$cloakedEntityClasses = @()
+	$vaultFile = $vault.DocumentService.GetFileById($file.Id)
+	$vaultFile = $propertySync.SyncProperties($vault, $vaultFile, "Property Sync", $true, [ref]$writeResults, [ref]$cloakedEntityClasses, $true)
+	Write-Host "End 'Sync Properties' of file '$($file.Name)'"
+}
+function UpdateRevisionBlock {
+	param ($file)
+
+	#using Vault API to get the file instead of using powerVault's $file since it runs into property sync job not finding any equivalence problems when '$file.Name' or '$file._Name' is used
+	$fileForUpdRevJob = $vault.DocumentService.GetLatestFileByMasterId($file.MasterId)
+
+	Write-Host "Start 'Update Revision Block' of file '$($fileForUpdRevJob.Name)' ..."
+
+	$ConnectivityJobProcessorDelegateAssembly = [System.Reflection.Assembly]::Load("Connectivity.JobProcessor.Delegate")
+	$context = $ConnectivityJobProcessorDelegateAssembly.CreateInstance("Connectivity.JobHandlers.Services.Objects.ServiceJobProcessorServices",$true, [System.Reflection.BindingFlags]::CreateInstance, $null, $null, $null, $null)
+	$context.GetType().GetProperty("Connection").SetValue($context, $vaultConnection, $null) #need to set the Connection property on the context or else it runs into error
+
+	$JobHandlerURBAssembly = [System.Reflection.Assembly]::Load("Connectivity.Explorer.JobHandlerUpdateRevisionBlock")
+	$uRBJobHandler = $JobHandlerURBAssembly.CreateInstance("Connectivity.Explorer.JobHandlerUpdateRevisionBlock.UpdateRevisionBlockJobHandler",$true, [System.Reflection.BindingFlags]::CreateInstance, $null, $null, $null, $null)
+	$uRBJob = New-Object Connectivity.Services.Job.UpdateRevisionBlockJob("vault",$fileForUpdRevJob.Id,$false,$false,$fileForUpdRevJob.Name)
+
+	$jobOutcome = $uRBJobHandler.Execute($context,$uRBJob) #call execute to start running the job
+
+	if ($jobOutcome -eq "Failure")
+	{
+		throw "Failed job 'Update Revision Block'" #Failed because of issue that occured in the job
+	}
+	Write-Host "End 'Update Revision Block' of file '$($fileForUpdRevJob.Name)'"
+}
```

Comment:
the one from the metso-minerals is an insteresting alternative to this block of code. maybe we should replace this one with the other one.